### ISSUE TITLE
Implement and Advertise Snippet Support

### ISSUE THREAD

**Issue by wbinnssmith** [atom/atom-languageclient#31]
Issue-izing the TODO found at [the autocomplete adapter](https://github.com/atom/atom-languageclient/blob/2b3ac65e87cb2832726e61502f1a2ab9d601fe6f/lib/adapters/autocomplete-adapter.js#L32)

When snippet support is implemented, the client should should also advertise this via the client capabilities sent to each language server.

**Comment by damieng** [comment]
Right now we're targeting v2 of the spec (v3 is still in flux) so we can skip the client capabilities part for now.

**Comment by wbinnssmith** [comment]
oh! didn't even realize client capabilities was part of v3.

it looks like v2 has a distinct `Snippet` completion type (which v3 has as well), but v3 (in its current form at least) has the option to use a snippet as the `insertTextFormat` for non-Snippet completion types. Looks like that's where having the client advertise its support for snippets is most useful as the server can decide the appropriate `insertTextFormat` to send back.

In the case of Flow, we can provide function completions with snippet text that expand to fill in the function's arguments, so this will be useful when we support v3.

**Comment by rcjsuen** [comment]
> v3 is still in flux

I thought v3 came out in [December 2016](https://github.com/Microsoft/language-server-protocol/commit/164eb31bd49535dca046034602146f96bb17b56d). Am I misunderstanding something?

**Comment by damieng** [comment]
https://github.com/Microsoft/language-server-protocol/commits/master/protocol.md shows there were changes all through May.

We'll add support for the v3 features in a future update.

**Comment by rcjsuen** [comment]
I don't believe those are breaking changes but certainly better safe than sorry.

**Comment by damieng** [comment]
v2 part is now done.

**Comment by damieng** [comment]
@wbinnssmith I think this has now been implemented through various incremental commits - can you confirm?

**Comment by UziTech** [comment]
Development of [atom-languageclient](https://www.npmjs.com/package/atom-languageclient) has officially moved to https://github.com/atom-ide-community/atom-languageclient 🎉 

If this is still an issue please consider opening an issue on that repo.
